cinema/views.py

```diff
@@ -48,6 +84,29 @@ class MovieSessionViewSet(viewsets.ModelViewSet):
     queryset = MovieSession.objects.all()
     serializer_class = MovieSessionSerializer
 
+    def get_queryset(self):
+        queryset = self.queryset
+
+        movie = self.request.query_params.get("movie")
+        date = self.request.query_params.get("date")
+
+        if movie:
+            queryset = queryset.filter(movie=movie)
+        if date:
+            queryset = queryset.filter(show_time__date=date)
+
+        if self.action == "list":
+            queryset = (
+                queryset.select_related("movie", "cinema_hall").annotate(
+                    tickets_available=(F("cinema_hall__rows")
+                                       * F("cinema_hall__seats_in_row")
+                                       - Count("tickets")
+                                       )
+                )
+            )
+
+        return queryset
```

Comment:
Good, but, as for me, it's more preferable to make calculated field, which we can use on all layers.